parquet-column/src/test/java/org/apache/parquet/schema/TestPrimitiveComparator.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.schema;
+
+import org.apache.parquet.io.api.Binary;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+
+import static org.apache.parquet.schema.PrimitiveComparator.BOOLEAN_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.DOUBLE_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.FLOAT_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.LEXICOGRAPHICAL_BINARY_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.BINARY_AS_SIGNED_INTEGER_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.SIGNED_INT32_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.SIGNED_INT64_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.UNSIGNED_INT32_COMPARATOR;
+import static org.apache.parquet.schema.PrimitiveComparator.UNSIGNED_INT64_COMPARATOR;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class TestPrimitiveComparator {
+
+  @Test
+  public void testBooleanComparator() {
+    boolean[] values = {false, true};
```

Comment:
I think this is too complicated for a boolean test with either 4 or 9 cases (true/false/null). It would be smaller and more clear to use `assertTrue(comparator.compare(true, false) > 0)`.

Be sure to also test null values for the boxed compare method.

Reply:
By adding `null` it is actually 9 comparisons to test (`null, null`, `null, true`, `null, false`, `true, null` etc.) so, I would keep the current pattern.
I'll extend the tests with `null`s.